LEAF_Nexus/sources/Data.php

```diff
@@ -74,6 +74,10 @@ public function getDataTableCategoryID()
      */
     public function getAllData($UID, $indicatorID = 0)
     {
+        if(!is_numeric($UID) || !is_numeric($indicatorID)) {
```

Comment:
The assertion of `!is_numeric($UID)` should be removed since there is no guarantee that all classes that inherit from `Data.php` will have numeric UIDs.